src/screens/drive/DrivePreviewScreen/DrivePreviewScreen.tsx

```diff
@@ -0,0 +1,248 @@
+  const [generatedThumbnail, setGeneratedThumbnail] = useState<GeneratedThumbnail>();
+  const dimensions = useWindowDimensions();
+  // REDUX USAGE STARTS
+  const insets = useSafeAreaInsets();
+  const dispatch = useAppDispatch();
+  // REDUX USAGE ENDS
+  const { downloadingFile } = useAppSelector((state) => state.drive);
+  // Use this in order to listen for state changes
+  const { focusedItem } = useAppSelector((state) => state.drive);
+  const totalHeaderHeight = DRIVE_PREVIEW_HEADER_HEIGHT + insets.top;
+  const topbarYPosition = useRef(new Animated.Value(0)).current;
+  const pdfViewerSpacerHeight = useRef(new Animated.Value(0)).current;
+
+  useEffect(() => {
+    if (
+      downloadingFile?.downloadedFilePath &&
+      VIDEO_PREVIEW_TYPES.includes(downloadingFile.data.type as FileExtension) &&
+      !generatedThumbnail
+    ) {
+      imageService
+        .generateVideoThumbnail(downloadingFile.downloadedFilePath)
+        .then((generatedThumbnail) => {
+          setGeneratedThumbnail(generatedThumbnail);
+        })
+        .catch((err) => errorService.reportError(err));
+    }
+  }, [downloadingFile?.downloadedFilePath]);
+
+  useEffect(() => {
+    Animated.timing(topbarYPosition, {
+      toValue: topbarVisible ? 0 : -totalHeaderHeight,
+      duration: 150,
+      useNativeDriver: true,
+      easing: DEFAULT_EASING,
+    }).start();
+
+    Animated.timing(pdfViewerSpacerHeight, {
+      toValue: topbarVisible ? totalHeaderHeight : 0,
+      duration: 100,
+      useNativeDriver: false,
+      easing: DEFAULT_EASING,
+    }).start();
+  }, [topbarVisible]);
+
+  const handleActionsButtonPress = () => {
+    dispatch(uiActions.setShowItemModal(true));
+  };
+
+  if (!downloadingFile) {
+    return <></>;
+  }
+
+  if (!focusedItem) {
+    return <></>;
+  }
+  const filename = `${focusedItem.name || ''}${focusedItem.type ? `.${focusedItem.type}` : ''}`;
+  const currentProgress = downloadingFile.downloadProgress * 0.5 + downloadingFile.decryptProgress * 0.5;
+  const FileIcon = getFileTypeIcon(focusedItem.type || '');
+  const hasImagePreview = IMAGE_PREVIEW_TYPES.includes(downloadingFile.data.type as FileExtension);
+  const hasVideoPreview = VIDEO_PREVIEW_TYPES.includes(downloadingFile.data.type as FileExtension);
+  const hasPdfPreview = PDF_PREVIEW_TYPES.includes(downloadingFile.data.type as FileExtension);
+  const getProgressMessage = () => {
+    if (!downloadingFile) {
+      return;
+    }
+
+    const progressMessage = strings.formatString(
+      currentProgress < 0.5 ? strings.screens.drive.downloadingPercent : strings.screens.drive.decryptingPercent,
+      (currentProgress * 100).toFixed(0),
+    );
+
+    return progressMessage;
+  };
+
+  const handleOpenFileWith = async (filename: string, downloadedFilePath: string) => {
+    try {
+      await fs.shareFile({
+        title: filename,
+        fileUri: downloadedFilePath,
+      });
+    } catch (error) {
+      errorService.reportError(error);
```

Comment:
This reports just to Sentry or is notified via UI too?

Reply:
Just to Sentry, adding UI feedback